### ISSUE TITLE
Installation problem

### ISSUE THREAD

**Issue by lukego** [styx-static/styx#31]
I have tried to install the latest version of Styx on a NixOS machine and it is not working for me. I suspect the issue is that at runtime styx is invoking nix with the system default nixpkgs (`import <nixpkgs>`) that does not match the nixpkgs that I installed styx from?

Here is what I am seeing:

```
[luke@interlaken:~]$ nix-env -i styx $(nix-build https://github.com/styx-static/styx/archive/latest.tar.gz)
error: anonymous function at /nix/store/vizrqzrx9qrrwn03fv9r379fcl4b46js-latest.tar.gz/derivation.nix:1:1 called without required argument ‘caddy’, at /nix/store/mqk7i2wcx521gacigcadsiq5i4lm1rvr-nixos-16.03.1349.dda40aa8d1/nixos/lib/customisation.nix:56:12
warning: name collision in input Nix expressions, skipping ‘/home/luke/.nix-defexpr/channels_root/nixos’
warning: there are multiple derivations named ‘styx-0.6.0’; using the first one
replacing old ‘styx-0.6.0’
installing ‘styx-0.6.0’

[luke@interlaken:~]$ rm -rf my-site

[luke@interlaken:~]$ styx new site my-site
Styx site initialized in '/home/luke/my-site'.

[luke@interlaken:~]$ cd my-site/

[luke@interlaken:~/my-site]$ styx preview
error: anonymous function at /home/luke/my-site/site.nix:6:1 called without required argument ‘styx’, at /nix/store/mqk7i2wcx521gacigcadsiq5i4lm1rvr-nixos-16.03.1349.dda40aa8d1/nixos/lib/customisation.nix:56:12
---
Error: Site could not been built, fix the errors and run the command again.
The '--show-trace' flag can be used to show debug information.
```

**Comment by ericsagnes** [comment]
This is indeed a design mistake, and I am afraid there is no easy solution for this problem at the moment.

A nasty way to "fix" this:

- Since 0.6.0 styx source includes a [nixpkgs](https://github.com/styx-static/styx/tree/v0.6.0/nixpkgs) folder which reference itself and themes.
- So one way to address this would be to clone locally the version you are using, and use styx with `--arg pkgs "/path/to/styx-0.6.0/nixpkgs"` (or use some shell magic that would not require cloning that I couldn't figure)

This problem should and **will** be fixed in the next version by having styx referring to itself and styx-themes from its own store path.

**Comment by ericsagnes** [comment]
Another way, probably easier to put in place is to pin styx and themes version directly in `site.nix` without relying on the passed arguments.

**Comment by ericsagnes** [comment]
Fixed in c0359dfeacdc56017c4f43adf636fe0ec3f2f239.
I plan to do the next release before the end of the month, but if you are in a hurry you can use the `master` version.
And if you can share your `site.nix` I could show some workarounds.

**Comment by lukego** [comment]
Thanks. I will come back to this later. Just now my "site" is actually only one document so I just used pandoc ([nix expr](https://github.com/lukego/studio/blob/manual/doc/default.nix) produces [html doc](https://hydra.snabb.co/build/1693306/download/1/studio.html)). However I will revisit the issue when I want to expand it, make it reactive, etc.
